Source/Workers/decodeDraco.js

```diff
@@ -53,33 +55,64 @@ define([
                 var attribute = dracoDecoder.GetAttributeByUniqueId(dracoGeometry, compressedAttribute);
                 var numComponents = attribute.num_components();
 
-                if (attribute.data_type() === 4) {
+                var i;
+                var quantization;
+                var transform = new draco.AttributeQuantizationTransform();
+                if (transform.InitFromAttribute(attribute)) {
+                    var minValues = new Array(numComponents);
+                    for (i = 0; i < numComponents; ++i) {
+                        minValues[i] = transform.min_value(i);
+                    }
+
+                    quantization = {
+                        quantizationBits : transform.quantization_bits(),
+                        minValues : minValues,
+                        range : transform.range(),
+                        octEncoded : false
+                    };
+                }
+                draco.destroy(transform);
+
+                transform = new draco.AttributeOctahedronTransform();
+                if (transform.InitFromAttribute(attribute)) {
+                    quantization = {
+                        quantizationBits : transform.quantization_bits(),
+                        octEncoded : true
+                    };
+                }
+                draco.destroy(transform);
+
+                var vertexArrayLength = numPoints * numComponents;
+                if (defined(quantization)) {
                     attributeData = new draco.DracoInt32Array();
-                    // Uint16Array is used becasue there is not currently a way to retreive the maximum
+                    vertexArray = new Int16Array(vertexArrayLength);
+                    dracoDecoder.GetAttributeInt32ForAllPoints(dracoGeometry, attribute, attributeData);
```

Comment:
When the attribute is oct encoded I think we can use a smaller array length since the number of components should be 2 instead of 3.

Also quantization should use a `Uint16Array` while oct-encoding should use a `Int16Array`.

Reply:
When the attribute is oct encoded, the loader already return 2 for the number of components, so the array length is being set to the correct length. 

I have updated `Uint16` vs `Int16` however.

Reply:
Ah ok, I should have checked that myself. Thanks!